test/custom-task-ctrls/wait-task/pkg/reconciler/reconciler_test.go

```diff
@@ -305,12 +333,23 @@ spec:
   - name: duration
     value: %s
 `, runName, tc.retries, tc.timeout, apiVersion, kind, tc.duration)
+			if tc.currentStatus != "" {
+				runYAML = runYAML + tc.currentStatus
+			}
 			r := parse.MustParseRun(t, runYAML)
 			if tc.isCancelled {
 				r.Spec.Status = v1alpha1.RunSpecStatusCancelled
 			}
+			if tc.startTime != nil {
+				r.Status.StartTime = tc.startTime
+			}
 
-			fakeReconcile(t, ctx, rec, r)
+			err := rec.ReconcileKind(ctx, r)
+			if err != nil {
+				if !strings.Contains(err.Error(), "requeue") {
```

Comment:
what's the reason that requeue errors are ignored? could you add a comment?

Reply:
This seems a bit weird to me-- returning a requeue error is not the same as returning successfully; why do the tests treat them the same?

Reply:
Ignoring the requeue error because we are testing a single round of reconciliation given Run Spec and Status. 

If we want to deal with the requeue event, then we need to duplicate some logics of deciding when to requeue, as discussed here: https://github.com/tektoncd/pipeline/pull/5435#discussion_r964113450.

Or do you mean we should also compare the error with expected error?

Reply:
Correct me if I'm wrong... as far as I understand, requeuing means we need to reconcile it again at some time, meaning there's actually no other runtime error.

Reply:
fwiw, in, say, `pkg/reconciler/pipelinerun/pipelinerun_test.go`, we do this too, albeit in a slightly different way:
https://github.com/tektoncd/pipeline/blob/0484584c518de3497dac5fa680888fdbc07683e5/pkg/reconciler/pipelinerun/pipelinerun_test.go#L2311

Reply:
ah ok-- good to know